chrome/popup.js

```diff
@@ -117,11 +142,127 @@ function deleteHosts(hostlist){
 }
 
 function openOptions() {
-  
-}
 
+}
 
+/*
 document.getElementById('button-options')
             .addEventListener('click', function() {
               chrome.tabs.create({ 'url': 'chrome://extensions/?options=' + chrome.runtime.id });
-          });
\ No newline at end of file
+          });
+*/
+
+var getRequestsDatabaseAdapter;
+async function loadRequestInfo() {
+  const databaseAdapter = await getRequestsDatabaseAdapter();
+
+
+  const domainList = document.getElementById("domainlist");
+  const checkedDomains = [];
```

Comment:
Some times not all domains are shown in the popus, I wonder if it has to do with how and when this checkedDomains set is (re-)created.

Reply:
I will check this next. So you can keep the PR open until it's fixed or merge it, what you prefer. But a good indication, thanks!

Reply:
I think this is solved. We should merge this branch before HotNets if we want to reference the code in the paper